superset/assets/javascripts/SqlLab/components/SqlEditor.jsx

```diff
@@ -208,7 +149,13 @@ class SqlEditor extends React.PureComponent {
       <div className="sql-toolbar clearfix">
         <div className="pull-left">
           <Form inline>
-            {runButtons}
+            <RunQueryActionButton
+              queryEditor={this.props.queryEditor}
```

Comment:
this is a good point. i agree passing in just the props the component needs is ideal. i'll make this change.